crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -196,8 +198,13 @@ impl<Param: SystemParam> SystemState<Param> {
         self.world_id == world.id()
     }
 
-    fn validate_world_and_update_archetypes(&mut self, world: &World) {
+    #[inline]
+    fn validate_world(&mut self, world: &World) {
         assert!(self.matches_world(world), "Encountered a mismatched World. A SystemState cannot be used with Worlds other than the one it was created with.");
+    }
+
+    #[inline]
+    pub fn update_archetypes(&mut self, world: &World) {
```

Comment:
This should have docs explaining when this needs to be called manually, and when the user doesn't need to worry about it.

Reply:
And what this method generally *does*.